lib/LaTeXImage.pm

```diff
@@ -86,6 +99,11 @@ sub tikzLibraries {
 # element the package options).
 sub texPackages {
 	my $self = shift;
+	# Legacy: if tikzpicture is the environment, make sure tikz package is included
+	if ($self->environment->[0] eq 'tikzpicture') {
+		return &$self('texPackages', ['tikz',@$_[0]]) if ref($_[0]) eq "ARRAY";
+		return &$self('texPackages', ['tikz']);
+	}
```

Comment:
Should there be any attempt to avoid loading tikz twice? We will automatically load it if `environment->[0] eq 'tikzpicture'`. But the author might include `tikz` in `texPackages()` too. The complication is they might load it as a string or as an array ref that includes package options. So it's a little harder to check if the author is explicitly loading it too.

If it means anything, `environment` only exists so that `PGtikz.pl` can use it to maintain automatic wrapping of the `tikzpicture` environment. If I were making this from scratch, I would just expect authors to write `\begin{...}...\end{...}` in the `tex()` part. So I don't _expect_ authors to use `environment` directly. But it's there, so they might.

Reply:
It would not be a bad idea to check to avoid loading it twice.

Reply:
I did as you suggested, and then also implemented a check for if `tikz` is already used. Maybe you can see a simpler  way. I turned the array of texPackages (a mix of strings and array refs) into a hash. Each key is a tex package name the author provided. The value from each key is 0 or 1, depending on if it matches 'tikz'. Then the check is boolean on `$istikzused{'tikz'}`. The result will either be undefined if the author did not include `tikz`, or 1 if they did.